### PR TITLE
chore: Creates internal drawer component

### PR DESCRIPTION
**Pull request by katiegeorge**
### Description

This is the first step to creating a Drawer component, which will release along with the public release of the drawers changes to the AppLayout. The goal of this PR is to get the code reviewed so that when the time comes to release the component publicly, it is mostly a "flip of a switch" to move the component out of the `internal` folder.

The purpose of the Drawer component is to provide a wrapper for the contents of a drawer, similar to the HelpPanel in the current tools slot or the SplitPanel in the splitPanel slot. It is built as a clone of the HelpPanel component, without the style overrides.

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/drawer/interfaces.ts
Comment (jperals) [inline src/internal/components/drawer/interfaces.ts:8]: Have these doc strings ben reviewed by content design?

File: src/internal/components/drawer/interfaces.ts
Comment (jperals) [inline src/internal/components/drawer/interfaces.ts:21]: "the panel" or "the drawer"?

File: src/internal/components/drawer/interfaces.ts
Comment (jperals) [inline src/internal/components/drawer/interfaces.ts:26]: "the panel" or "the drawer"?

File: src/internal/components/drawer/index.tsx
Comment (just-boris) [reply src/internal/components/drawer/index.tsx:17]: This is the current convention. All components inside the components repository need to support it for consistency. 

New packages (board-components and any other future one) do not have to do this

File: src/i18n/messages-types.ts
Comment (just-boris) [inline src/i18n/messages-types.ts:160]: What does `never` mean here? Is it temporary while the strings are not avaialble?

File: src/internal/components/drawer/interfaces.ts
Comment (katiegeorge) [reply src/internal/components/drawer/interfaces.ts:21]: I'll updated to drawer.

File: src/internal/components/drawer/interfaces.ts
Comment (katiegeorge) [reply src/internal/components/drawer/interfaces.ts:8]: They are copied word for word from the Help Panel documentation, other than `drawer` instead of `panel`. So they were reviewed when the help panel was released.

File: src/i18n/messages-types.ts
Comment (katiegeorge) [reply src/i18n/messages-types.ts]: If you look at this file, most of the types are `never`. Although I missed that this file is auto-generated, so I removed this update in this PR.

**Comment by katiegeorge** [comment]
Realizing I need to update the `i18n` package with the updated drawer strings before merging this in. Moving to a draft PR until that is merged.

File: src/internal/components/drawer/styles.scss
Comment (jperals) [inline src/internal/components/drawer/styles.scss:13]: Why does the help panel have zero bottom padding and the drawer `$space-scaled-xxxl`?

File: src/internal/components/drawer/styles.scss
Comment (jperals) [inline src/internal/components/drawer/styles.scss:17]: Would it make sense to de-duplicate this block?

**Comment by jperals** [comment]
Do we want to add the new testing utilities as well? I am guessing we could add them to `src/test-utils/dom/internal/drawer`. And then you could use them in your tests instead of `findByClassName`.

File: src/internal/components/drawer/styles.scss
Comment (katiegeorge) [reply src/internal/components/drawer/styles.scss:13]: Help panel has built-in padding for the last item (see [here](https://github.com/cloudscape-design/components/blob/main/src/help-panel/styles.scss#L76)), but since Drawers won't be content-based, the padding goes on the container.

File: src/internal/components/drawer/styles.scss
Comment (katiegeorge) [reply src/internal/components/drawer/styles.scss:17]: Do you mean make it the same for help panel and drawer? The header scss is already different, since the drawer header doesn't have the same style overrides as help panel. And I don't want to overlap them too much, since they may diverge even more in the future (ie, there are talks of putting other action buttons in the drawer header).

**Comment by katiegeorge** [comment]
> Do we want to add the new testing utilities as well? I am guessing we could add them to `src/test-utils/dom/internal/drawer`. And then you could use them in your tests instead of `findByClassName`.

My plan is to add the test utilities when we make this public. The API review document includes `findHeader` and `findContent` just like the Help Panel, so that's the plan going forward.

File: src/internal/components/drawer/styles.scss
Comment (jperals) [reply src/internal/components/drawer/styles.scss:17]: The CSS rules in this `.header` selector look exactly the same to me as in help panel (see [here](https://github.com/cloudscape-design/components/blob/main/src/help-panel/styles.scss#L107)) with the only exception that the rule `border: 0` is missing here in the drawer —I assume you removed it because it's actually unnecessary (and it seems unnecessary in the help panel too).

Not sure how much if at all adding action buttons in the drawer header would need to affect this block, with this information I would personally still be inclined to deduplicate, but won't block the PR on this.

**Review by jperals — APPROVED** [review]

**Pull request merged**
